### ISSUE TITLE
[Help]: About s2a in maskgct and dualcodec

### ISSUE THREAD

**Issue by MrWaterZhou** [open-mmlab/Amphion#469]
I have a few questions about training the **S2A module** in **MaskGCT**, specifically around the acoustic token prediction setup and the training behavior I’m observing with DualCodec.

1. When predicting acoustic tokens, the first layer and the other layers are modeled with two separate models. What is the motivation or design rationale behind separating the first layer from the remaining layers?

2. I trained S2A using **DualCodec**, roughly following the batch size and learning rate settings from the paper. During training, both the loss and accuracy do not look very good by the end (loss stays around **4.xx**, accuracy around **0.1xx**), but the actual reconstruction quality sounds quite good. Is this behavior expected or normal?

<img width="1775" height="174" alt="Image" src="https://github.com/user-attachments/assets/6a3d3c77-eb62-4b65-b2b6-85154b342c1a" />

**Comment by jiaqili3** [comment]
(2) is normal. The accuracy of speech token is usually lower, especially with DualCodec's 16k+ codebook size.

**Comment by MrWaterZhou** [comment]
> (2) is normal. The accuracy of speech token is usually lower, especially with DualCodec's 16k+ codebook size.

Thanks for your reply. I have one more question.
Although the final results sound good and the objective metrics meet expectations, I noticed that the attention_mask during inference seems to be all zeros. I’m a bit confused about this.
https://github.com/open-mmlab/Amphion/blob/d3de90f02e7f1c7dbc24b04d52fe1b8ef438effe/models/tts/maskgct/llama_nar.py#L256C1-L294C1

**Comment by jiaqili3** [comment]
Thanks for asking, however, I’m not familiar with the code you’re referring to. Maybe all zero mask means no masking to the sequence (attend to all positions).
